### PR TITLE
 proc_macro_deps is its own attribute

### PR DESCRIPTION
**Pull request by illicitonion**
This allows an unconditional transition to the `exec` config, rather
than needing a conditional transition (which cannot transition to the
`exec` config, only to the `host` config).

See https://github.com/bazelbuild/rules_rust/issues/327

### CODE REVIEW

File: impl/src/planning.rs [hunk 1]
```diff
@@ -581,29 +585,43 @@ impl<'planner> CrateSubplanner<'planner> {
...
+          proc_macro_deps.push(buildable_dependency);
+        } else {
+          normal_deps.push(buildable_dependency);
+          // Only add aliased normal deps to the Vec
+          if let Some(alias) = aliased_dep_names.get(&dep_package.name) {
+            aliased_deps.push(DependencyAlias {
+              target: buildable_target.clone(),
+              alias: util::sanitize_ident(alias),
+            })
+          }
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply impl/src/planning.rs]: Done

File: impl/src/planning.rs [hunk 2]
```diff
@@ -581,29 +585,43 @@ impl<'planner> CrateSubplanner<'planner> {
...
     })
   }
 
+  fn is_proc_macro(&self, package: &Package) -> bool {
+    package
+      .targets
+      .iter()
+      .flat_map(|target| target.crate_types.iter())
+      .any(|crate_type| crate_type.as_str() == "proc-macro")
+  }
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply impl/src/planning.rs]: Moved to `BuildableDependency`, but added a lengthy comment about why we're doing a Package-wide lookup.

File: impl/src/templates/partials/rust_target.template [hunk 3]
```diff
@@ -1,13 +1,31 @@
+{%- if target.kind != "bin" and target_name_sanitized != crate_name_sanitized %}
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply impl/src/templates/partials/rust_target.template]: Done

**Comment by illicitonion** [comment]
Added a unit test :) Thanks!

**Comment by dfreese** [comment]
Thanks!  Looks good, though it looks like you'll need to bump rules rust in the smoke test to include your proc_macro_deps change.

**Pull request merged**